Tests/Integration/Network/Connection/FogSyncErrorConnectionIntTests.swift

```diff
@@ -0,0 +1,266 @@
+    
+    func fogBehindConsensusAtThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 60
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold - 1
+        
+        // Fog behind at threshold, should fail
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: false
+        )
+    }
+    
+    func testFogBehindConsensusOverThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusOverThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusOverThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 410
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold
+        
+        // Fog behind over threshold, should fail
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: false
+        )
+    }
+    
+    func testViewLedgerAndConsensusWithinThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try viewLedgerAndConsensusWithinThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func viewLedgerAndConsensusWithinThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let viewIndex: UInt64 = 234234235675
+        let ledgerIndex = viewIndex - MockFogSyncChecker.threshold + 2
+        let consensusIndex = viewIndex
+        
+        // below threshold, should pass
+        try attemptRefresh(
+            fogViewBlockIndex: viewIndex,
+            fogLedgerBlockIndex: ledgerIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testViewLedgerAndConsensusAtThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try viewLedgerAndConsensusAtThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func viewLedgerAndConsensusAtThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let viewIndex: UInt64 = 99
+        let ledgerIndex = viewIndex + MockFogSyncChecker.threshold + 1
+        let consensusIndex = viewIndex + MockFogSyncChecker.threshold / 2
```

Comment:
Isn't this really within threshold,  instead of at threshold?  At threshold makes me think they should be both set to viewIndex + threshold